dali/dfu/dfurun.cpp

```diff
@@ -1330,7 +1337,7 @@ class CDFUengine: public CInterface, implements IDFUengine
                                     dstCompressed = srcFile->isCompressed();
                                 else
                                 {
-                                    IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : foreignfdesc.get());
+                                    IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : srcFile->getFileDescriptor());
                                     dstCompressed = srcDesc && srcDesc->isCompressed();
                                 }
```

Comment:
if srcFile were non null it cannot be here..
So I think should change these lines to:
```
   else if (auxfdesc)
       dstCompressed = auxfdesc->isCompressed();
```

Reply:
Changed